lib/Zonemaster/CLI.pm

```diff
@@ -517,13 +518,24 @@ sub add_fake_delegation {
         if ($ip) {
             push @{ $data{ $self->to_idn( $name ) } }, $ip;
         }
+        elsif (Zonemaster::Zone->new( { name => $domain } )->is_in_zone( $name )) {
+            say STDERR "--ns in zone nameservers must be a name/ip pair.";
+            exit( 1 );
+        }
         else {
             my $n = $self->to_idn( $name );
             my @ips = Net::LDNS->new->name2addr($n);
-            push @{ $data{$n} }, $_ for @ips;
+            if ( @ips ) {
+                push @{ $data{$n} }, $_ for @ips;
+            }
+            else {
+                say STDERR "--ns ${name} nameserver has no IP address.";
+                exit( 1 );
+            }
```

Comment:
Actually, this issue highlights a design problem, rather than a bug, in the cli (and the backend). Can you guys bring this to the attention of the entire dev team, so that everyone is on the same page? And then see how you can rework both the cli and backend to do the right thing and be consistent.